core/include/detray/intersection/intersection.hpp

```diff
@@ -131,60 +130,4 @@ struct intersection2D {
     }
 };
 
-/// @brief This class holds the intersection information, including the global
-/// point of intersection.
-///
-/// @tparam surface_descr_t is the type of surface descriptor
-template <typename surface_descr_t,
-          typename algebra_t = __plugin::transform3<detray::scalar>>
-struct intersection2D_point
-    : public intersection2D<surface_descr_t, algebra_t> {
-    using scalar_type = typename algebra_t::scalar_type;
-    using point3 = typename algebra_t::point3;
-    using point2 = typename algebra_t::point2;
-
-    /// Intersection point in global 3D cartesian coordinate frame
-    point3 p3{detail::invalid_value<scalar_type>(),
-              detail::invalid_value<scalar_type>(),
-              detail::invalid_value<scalar_type>()};
-
-    /// Transform to a string for output debugging
-    DETRAY_HOST
-    friend std::ostream &operator<<(std::ostream &out_stream,
-                                    const intersection2D_point &is) {
-        scalar_type r{getter::perp(is.p3)};
-        out_stream << "dist:" << is.path << " [glob: r:" << r
```

Comment:
I am not particularly happy with removing the printout of the global point from this code. It is quite helpful to know where in the detector something is breaking/overlapping. For example, I can see in the navigation inspector output from the radius of the intersection which portal cylinder, beampipe etc. got intersected. I am actually relying on this information quite a bit when I do navigation debugging. Can we add this information back to the output somehow, even if it is not in the intersection?

This is an example printout of what I need for debugging (note, that the surface index is much harder to interpret, as it is in general also dependent on how many barrel/endcap layers I add to the toy detector and I don't know all of the indices by heart :)):
```
Init complete: 
Volume				0
No. reachable		2
Surface candidates: 
dist:261.823 [glob: r:19, z:261.133 | loc: 59.6903, 261.133], (sf index:vol = 0 | id = 2 | index = 0 | extra = 255, links to vol:0), status: inside, direction: along
dist:372.065 [glob: r:27, z:371.084 | loc: 84.823, 371.084], (sf index:vol = 0 | id = 0 | index = 15 | extra = 255, links to vol:15), status: inside, direction: along
=> next:  -> vol = 0 | id = 2 | index = 0 | extra = 255
status				towards_surface
current object			undefined
distance to next		261.823
trust				full_trust
```
Currently, this is done by printing the intersection in the navigation inspector. If we can make the inspector print this information in another way, that would also work

Reply:
the navigator inspector takes the navigator state as an input if I am correct?
Then I can add a `local_to_global` function in detector_kernel.hpp and call it in the inspector operator - Does it sound OK to you?

Reply:
Yes, it does. Sounds good to me

Reply:
Yes, it does. That sounds good

Reply:
I found that it is not possible because I can not bring the direction information which is required for local_to_global transformation. any thought?

Reply:
and p2 of intersection is also in measurement frame so it may not be used for the global transformation I guess?

Reply:
Hm, right... Just leave the direction out for now? We will have to come up with something for the lines later. The only other mask that has a difference in measurement frame vs. local frame right now is the ```cylinder3D```, which is not used in the navigation. Otherwise, we might need to write a full propagation inspector at some point. I already have a PR open for a stepper inspector

Reply:
OK, I forgot the ```helix_cylinder_intersector```...

Reply:
I don't feel like to have an exception. Let me leave this PR open until I find a solution

Reply:
BTW, do we have to keep the navigator inspector when there is a propagator inspector?

Reply:
I don't know - It would be possible to promote navigator inspector and stepper inspector to `detray::actor` but it's still a pain in many aspects.

Reply:
Don't get me wrong, I think you are right, the separate ```intersection2D_point``` class is super ugly and we should get rid of it somehow

Reply:
I think all this problems occur due to the lack of information in `p2`.  It can not be converted to the global position even with direction for some coordinates.

My suggestion is:
1. Make `p2` have three components (for the most of coordinates like cartesian2, the last element will be empty. `p2` of line will have (L,z) -> (L, z, phi))
2. Unify the local and mesaurement frame because p2 can cover them hopefully...

If I am correct, this won't even require direction information to recover the global position from `p2`.

How do you think

Reply:
Actually I see this as an opportunity. Stay tuned...

Reply:
Now we can print global position of intersection objects:
```
Exited: 
Volume                          65535
No. reachable           0
Surface candidates: 
[glob: r:164, z:0 | loc: 128.805, 0] dist:-5.95465e-06, (sf index:vol = 13 | id = 1 | index = 2886 | extra = 255, links to vol:12), status: inside, direction: opposite
[glob: r:171.299, z:0 | loc: 3.48354, -35] dist:3.60212e-07, (sf index:vol = 13 | id = 0 | index = 2388 | extra = 255, links to vol:13), status: inside, direction: along
[glob: r:171.797, z:0 | loc: 3.49367, 35] dist:-3.72219e-06, (sf index:vol = 13 | id = 0 | index = 2310 | extra = 255, links to vol:13), status: inside, direction: opposite
[glob: r:180, z:0 | loc: 141.372, 0] dist:-0, (sf index:vol = 13 | id = 1 | index = 2887 | extra = 255, links to vol:65535), status: inside, direction: opposite
=> next: exhausted
status                          e_on_target
current object                  vol = 13 | id = 1 | index = 2887 | extra = 255
distance to next                on obj (within tol)
trust                           full_trust
```